libs/blocks/graybox-promote/components/graybox-promote-lit.js

```diff
@@ -48,80 +47,87 @@ const getSheetValue = (data, key) => data?.find(
   (obj) => obj.key?.toLowerCase() === key?.toLowerCase(),
 )?.value;
 
-const getAemInfo = () => {
+const getAemInfo = (context) => {
   const search = new URLSearchParams(window.location.search);
-  return {
-    ref: search.get('ref'),
-    repo: search.get('repo'),
-    owner: search.get('owner'),
-    excelRef: search.get('referrer'),
-  };
+  context.ref = search.get('ref');
+  context.repo = search.get('repo');
+  context.owner = search.get('owner');
+  context.excelRef = search.get('referrer');
 };
 
-const getProjectInfo = async (url) => {
-  const liveOrigin = url.origin.replace('.hlx.page', '.hlx.live');
+const getProjectInfo = async (context) => {
   const sheet = await getJson(
-    `${liveOrigin}${url.pathname}?sheet=settings`,
+    `${context.liveUrl.href}?sheet=settings`,
     'Failed to fetch project info',
   );
-  return {
-    experienceName: getSheetValue(
-      sheet.data,
-      KEYS.PROJECT_INFO.EXPERIENCE_NAME,
-    ),
-    grayboxIoEnv: getSheetValue(sheet.data, KEYS.PROJECT_INFO.GRAYBOX_IO_ENV),
-  };
+  context.setup.experienceName = getSheetValue(
+    sheet.data,
+    KEYS.PROJECT_INFO.EXPERIENCE_NAME,
+  );
+  context.grayboxIoEnv = getSheetValue(sheet.data, KEYS.PROJECT_INFO.GRAYBOX_IO_ENV);
 };
 
-const getGrayboxConfig = async ({ ref, repo, owner, grayboxIoEnv } = {}) => {
+const getGrayboxConfig = async (context) => {
+  const { ref, repo, owner, grayboxIoEnv } = context;
   const sheet = await getJson(
     `https://${ref}--${repo}--${owner}.hlx.live/.milo/graybox-config.json`,
     'Failed to fetch graybox config',
   );
 
   const grayboxData = sheet.graybox?.data;
-
-  return {
-    promoteDraftsOnly:
-      getSheetValue(
-        grayboxData,
-        KEYS.CONFIG.PROMOTE_DRAFTS_ONLY,
-      )?.toLowerCase() === 'true',
-    enablePromote: getSheetValue(grayboxData, KEYS.CONFIG.ENABLE_PROMOTE)?.toLowerCase() === 'true',
-    promoteUrl: getSheetValue(
-      grayboxData,
-      KEYS.CONFIG.PROMOTE_URL[(grayboxIoEnv || 'prod').toUpperCase()],
-    ),
-    promoteIgnorePaths: sheet.promoteignorepaths?.data
-      ?.map((item) => item?.[KEYS.CONFIG.PROMOTE_IGNORE_PATHS])
-      .join(','),
-  };
+  context.enablePromote = getSheetValue(grayboxData, KEYS.CONFIG.ENABLE_PROMOTE)?.toLowerCase() === 'true';
+  context.promoteUrl = getSheetValue(
+    grayboxData,
+    KEYS.CONFIG.PROMOTE_URL[(grayboxIoEnv || 'prod').toUpperCase()],
+  );
+  context.setup.draftsOnly = getSheetValue(
+    grayboxData,
+    KEYS.CONFIG.PROMOTE_DRAFTS_ONLY,
+  )?.toLowerCase() === 'true';
+  context.setup.promoteIgnorePaths = sheet.promoteignorepaths?.data
```

Comment:
is context.setup available?

Reply:
Is always available because is initialised in [here](https://github.com/adobecom/milo/pull/2584/files#diff-7214629f44f7a1dbc0bdfc18fb6309965da48000fe961e4aaeed56046e9aa06fR129).